sql-common/my_time.c

```diff
@@ -336,12 +339,24 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && (*pos == '-' || *pos == '+') && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
+  {
+    status->tz_offset_valid = str_to_offset(pos, 6, &status->tz_offset); //RFC3339 specifies the offset will be exactly 6 digits
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == 'Z' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuuZ is supported too */
+  {
+    status->tz_offset_valid = 1;
+    status->tz_offset = 0;
+    pos++;
+  }
+
 
   if (pos == end)
   {
     /*
       Found date in internal format
-      (only numbers like [YY]YYMMDD[T][hhmmss[.uuuuuu]])
+      (only numbers like [YY]YYMMDD[T|t][hhmmss[.uuuuuu]][[Z|z]|[[-|+]hh[[:]mm]]])
```

Comment:
This regex is the ISO8601 because it implies that minute offsets and the colon are optional as per its standard. This currently isn't the case in the way str_to_offset was implemented or the explicit 6 character length above.